src/realm/object-store/c_api/dictionary.cpp

```diff
@@ -86,6 +86,30 @@ RLM_API bool realm_dictionary_insert(realm_dictionary_t* dict, realm_value_t key
     });
 }
 
+RLM_API realm_object_t* realm_dictionary_insert_embedded(realm_dictionary_t* dict, realm_value_t key)
+{
+    return wrap_err([&]() {
```

Comment:
I guess this is wanted, the whole API seems to ignore the possibility that the ptr passed as argument can be `null`. I guess in this case, crashing is OK.